Missing FileNameFactories

I seem to have encountered a strange bug, where the FileNameFactories aren't loading properly.

I've imported the NuGet package into my project:

```xml
<ItemGroup>
  <PackageReference Include="DefaultDocumentation" Version="0.8.1" />
</ItemGroup>
```

I'm leaving all configuration set to default for now. But upon building the application, it throws (paths shortened):

```powershell
..\DefaultDocumentation.targets(45,3): error MSB4018: The "DefaultDocumentationTask" task failed unexpectedly. [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018: System.Exception: FileNameFactory 'FullName' not found [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at DefaultDocumentation.Internal.Context..ctor(JObject configuration, Type[] availableTypes) in D:\a\DefaultDocumentation\DefaultDocumentation\source\DefaultDocumentation.Common\Internal\Context.cs:line 20 [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at DefaultDocumentation.Internal.GeneralContext..ctor(JObject config, Type[] availableTypes, Settings settings, IReadOnlyDictionary`2 items) in D:\a\DefaultDocumentation\DefaultDocumentation\source\DefaultDocumentation.Common\Internal\GeneralContext.cs:line 30 [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at DefaultDocumentation.Generator..ctor(Target loggerTarget, IRawSettings settings) in D:\a\DefaultDocumentation\DefaultDocumentation\source\DefaultDocumentation.Common\Generator.cs:line 116 [..\Utilities.csproj]  
..\DefaultDocumentation.targets(45,3): error MSB4018:    at DefaultDocumentation.Generator.Execute(Target loggerTarget, IRawSettings settings) in D:\a\DefaultDocumentation\DefaultDocumentation\source\DefaultDocumentation.Common\Generator.cs:line 196 [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at DefaultDocumentation.DefaultDocumentationTask.Execute() in D:\a\DefaultDocumentation\DefaultDocumentation\source\DefaultDocumentation\DefaultDocumentationTask.cs:line 57 [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [..\Utilities.csproj]
..\DefaultDocumentation.targets(45,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [..\Utilities.csproj]
```

Poking around the source code, I can see in [`Generator.cs` line 118](https://github.com/Doraku/DefaultDocumentation/blob/84afa18196236f3997808486cb4a9c1a8c8938cd/source/DefaultDocumentation.Common/Generator.cs#L118) that it's supposed to be loading all the `availabletypes`:

```csharp
new[] { typeof(Markdown.Writers.MarkdownWriter).Assembly.Location }
    .Concat(GetSetting<string[]>(nameof(settings.Plugins)) ?? Enumerable.Empty<string>())
    .Select(Assembly.LoadFrom)
    .SelectMany(a => a.GetTypes())
    .ToArray()
```

... and then in [`Context.cs` line 19](https://github.com/Doraku/DefaultDocumentation/blob/84afa18196236f3997808486cb4a9c1a8c8938cd/source/DefaultDocumentation.Common/Internal/Context.cs#L19) it's sorting through them to find the specified `FileNameFactory`:

```csharp
string fileNameFactory = GetSetting<string>(nameof(IRawSettings.FileNameFactory));
FileNameFactory = string.IsNullOrEmpty(fileNameFactory) ? null : availableTypes
    .Where(t => typeof(IFileNameFactory).IsAssignableFrom(t) && !t.IsAbstract)
    .Select(t => (IFileNameFactory)Activator.CreateInstance(t))
    .LastOrDefault(f => f.Name == fileNameFactory || $"{f.GetType().FullName} {f.GetType().Assembly.GetName().Name}" == fileNameFactory)
    ?? throw new Exception($"FileNameFactory '{fileNameFactory}' not found");
```

But I can't tell where the code would be failing to find the `IFileNameFactory` types. It all looks good to me. There must be some edge case I've stumbled on, where it's not working. My repo is a monorepo structure, so it's a bit different than a standard repo layout; and perhaps something in the way it's configured is the culprit.

I need to track this down, but I don't know how to go about debugging it further. I'd be happy to work with you, if you have any ideas.

--- a/source/DefaultDocumentation.Common/Generator.cs
+++ b/source/DefaultDocumentation.Common/Generator.cs
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
 using System.IO;
 using System.Linq;
 using System.Linq.Expressions;
@@ -115,13 +115,16 @@ namespace DefaultDocumentation
 
             _context = new GeneralContext(
                 _configuration,
-                new[] { typeof(Markdown.Writers.MarkdownWriter).Assembly.Location }
-                    .Concat(GetSetting<string[]>(nameof(settings.Plugins)) ?? Enumerable.Empty<string>())
-                    .Select(Assembly.LoadFrom)
+                new[] { typeof(Markdown.Writers.MarkdownWriter).Assembly }
+                    .Concat(
+                        (GetSetting<string[]>(nameof(settings.Plugins)) ?? Enumerable.Empty<string>())
+                        .Select(Assembly.LoadFrom)
+                    )
                     .SelectMany(a => a.GetTypes())
                     .ToArray(),
                 resolvedSettings,
                 DocItemReader.GetItems(resolvedSettings));
+
         }
 
         private void WritePage(DocItem item, StringBuilder builder)

